src/rgw/rgw_rest_s3.cc

```diff
@@ -1303,12 +1303,19 @@ struct ReplicationConfiguration {
         return -EINVAL;
       }
 
+      if (std::holds_alternative<rgw_account_id>(s->owner.id)) {
```

Comment:
the `rgw_owner` variant may someday get additional types, so i'd prefer this check for "not rgw_user"

if we know the variant type is `rgw_user`, we could just read its tenant field directly with something like:
```c++
const auto& tenant_owner = std::get_if<rgw_user>->tenant;
```

Reply:
> the `rgw_owner` variant may someday get additional types, so i'd prefer this check for "not rgw_user"

:+1: 

> if we know the variant type is rgw_user, we could just read its tenant field directly

That doesn't work I'm afraid as `SysReqApplier` needs to kick in here and return the effective one than the user's one (as the `s->owner.id` is only `ACLOwner` and it doesn't hold the tenant to be accessed).